### DISCUSSION TITLE
Request: make Max Lens compatible with post stabilisation

### DISCUSSION THREAD

**Discussion by StevensonPrime** [gopro/labs#927] [Feature Requests]
I don't know if this is the appropriate place to ask this and I see there's been some similar requests in the main GoPro Forum too, but here's my little grain of sand.
Currently all GoPro models with Max Lens don't allow for using Max Lens in full resolution and FOV aspect ratio for post stabilization one way or another, the details are very technical but I'm sure GoPro knows all about it.
Thanks

**Comment by dnewman-gpsw** [comment]
Max Lens Mod is stabilizing to it maximum resolution in camera, given that the image circle mostly fits within the sensor. 5.3K from MLM is not possible. If it turn the MLM mode off, and work with the raw image circle data, you will have a lot of work to do. It is technically possible to post stabilize if you disable the ERS, yet will not gain resolution doing that, and the stabilization will not be quite as good as in camera. 

This is not a Labs question, although the Labs extension for BERS would be used.

**Comment by StevensonPrime** [comment]
Thanks for your rapid response, I understand that this is not a labs question.
Nonetheless some considerations about BERS and resolution:

- When using BERS, that is recording video in-camera with rolling shutter uncorrected, the next problem that I'm sure GoPro is aware is that to post correct that rolling shutter with software it needs a fairly good precise gyro data to be recorded for syncing with the video and currently GoPro does not so, instead, since Reelsteady does not support Max Lens fisheye, with a stabilization software like Gyroflow used by the cinema industry it seems many people have had the same problem of not being able to sync appropriately to correct the rolling shutter because of "bad gyro data" and I'm quoting the Gyroflow developer.
- Resolution wise when a fisheye image is undistorted to linear perspective the resolution density of the image is proportionally distorted leaving an image with even more resolution than the original image in the center and incrementally poor res in the periphery, BTW i believe human vision does that as well.
- It's true that this is gyro data problem but I was wondering if GoPro Labs is able to provide an alternative solution

**Reply by dnewman-gpsw** [reply]
1. The Gyro data has all the info there to have synced image and IMU, the only camera that had issues with RS Go (that only used GPMF Gyro data) was HERO7. Gyroflow dev has likely gotten spoiled by using using CORI (camera orientation) metadata, that you get with ERS enabled. Using raw gyro is way harder. The only issue with the gyro data is it frequency of 200Hz, limiting the amount rejection of high frequency jello. Yet RSGo works with 200Hz gyro, gyroflow should be able to do the same.

2. Then shoot Wide, and do a Linear conversion in post for the same benefit. 

3. It is not true.  If GF can't do it at 200Hz, there is no point in offering a higher frequency through Labs.

**Reply by StevensonPrime** [reply]
Thanks for the response, I meant "bad" raw gyro data not the default synchronized quaternion that GoPro uses per-frame, this works well with pre-rolling shutter corrected GoPro material but it's not enough to correct rolling shutter with millisecond accuracy for using BERS. Currently when syncing with raw gyro data looks like the millisecond accuracy gets distorted as the video gets longer but always the start and end of the recording maintains a consistent and correct, I assume, millisecond value.
Of course, I'm using all in manual exposure settings including the Labs code to disable Gyro shutter speed compensation.

**Reply by dnewman-gpsw** [reply]
I'm saying that the handholding of the quaternions is not how RSGo did it originally. We added that only to ease stabilization with Hypersmooth enabled (using CORI and ORIO together.) But the raw data is also good. All the data is there to stabilize and ERS correct using microseconds timestamps on gyro data and video metadata enabling precision sync. Hypersmooth uses the same, just at a higher frequency,. 200Hz will be okay up to 60fps video, for the most common camera movement. 

P.S. CORI is an output, not an input to Hypersmooth.
